R/forecast-bbs-core.R

```diff
@@ -299,17 +299,51 @@ add_observers=function(bbs_data){
                     bbs_weather.obsn as observer_id,
                     year,
                     (statenum*1000)+route as site_id
-                  FROM 
+                  FROM
                     bbs_weather
-                  WHERE 
+                  WHERE
                     runtype=1 AND rpid=101'
-  
+
   observer_info=db_engine(action = 'read', sql_query = observer_query)
-  
+
   bbs_data %>%
     left_join(observer_info, by=c('site_id','year'))
 }
 
+#' Add point estimates for random effects associated with site_id and observer_id
+#' @param df a data.frame, as produced by \link{get_bbs_data}
+#' @param last_training_year the final year to include in the training set for
+#'        fitting the mixed model
+#' @importFrom lme4 lmer ranef
+add_ranefs = function(df, last_training_year) {
+
+  df = add_observers(df)
```

Comment:
Any reason not to add this to `get_richness_ts_env_data` and then use the resulting dataframe as the starting point instead of recalculating richness again in a separate location?